src/main/java/com/customerApi/api/services/CustomerService.java

```diff
@@ -0,0 +1,110 @@
+package com.customerApi.api.services;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.customerApi.api.exception.CustomerNotFoundException;
+import com.customerApi.api.modals.Customer;
+import com.customerApi.api.repositories.CustomerRepository;
+
+@Service
+public class CustomerService {
+
+	private static final Logger logger = LoggerFactory.getLogger(CustomerService.class);
+
+	@Autowired
+	private CustomerRepository customerRepo;
+
+	public boolean addCustomerData(Customer customer) {
+		boolean status = false;
+		try {
+
+			logger.trace("Running post customer service");
+			customerRepo.save(customer);
+			logger.debug("Customer First Name {}, last name {}, sign-in in", customer.getFirstName(),
+					customer.getLastName());
+			status = true;
+		} catch (NullPointerException e) {
+			logger.error("Entered customer record is empty: " + e);
+		}
+
+		return status;
+
+	}
+
+	public List<Customer> getCustomersData() {
+		logger.trace("Running get customer service :");
+		return customerRepo.findAll();
+	}
+
+	public String getCustomerDataById(int id) {
+		logger.trace("Running get customerById service :");
+		Optional<Customer> customer = customerRepo.findById(id);
+
+		if (!customer.isPresent()) {
+
+			logger.error("Customer ID {}" + id);
+			throw new CustomerNotFoundException("id-" + id);
+
+		} else {
```

Comment:
Do you really think else is required here???

Reply:
Thanks for your feedback. I have removed this else condition. As if customer record is not present it will throw error rather it will delete the record.